File: Algorithms/Easy/Pascal'sTriangle.py
# Time complexity: O(n^2)
# Space complexity: O(1)
class Solution:
    def generate(self, numRows: int) -> list[list[int]]:
        triangle = [[1]]
        for i in range(numRows - 1):
            temp = 0
            row = []
            for j in range(len(triangle[i])):
                row.append(temp + triangle[i][j])
                temp = triangle[i][j]

            row.append(1)
            triangle.append(row)

        return triangle

File: Algorithms/Easy/118.Pascal's-Triangle/Pascal'sTriangle.go
package easy

// Time complexity: O(n^2)
// Space complexity: O(1)
func generate(numRows int) [][]int {
	triangle := make([][]int, 0, numRows)
	triangle = append(triangle, []int{1})

	for i := 0; i < numRows-1; i++ {
		row := make([]int, 0, len(triangle[i])+1)
		temp := 0

		for _, val := range triangle[i] {
			row = append(row, temp+val)
			temp = val
		}

		row = append(row, 1)
		triangle = append(triangle, row)
	}

	return triangle
}
